autogen/agentchat/conversable_agent.py

```diff
@@ -74,25 +78,24 @@ class ConversableAgent(LLMAgent):
     def __init__(
         self,
         name: str,
-        system_message: Optional[Union[str, List]] = "You are a helpful AI Assistant.",
+        system_message: str = "You are a helpful AI Assistant.",
         is_termination_msg: Optional[Callable[[Dict], bool]] = None,
-        max_consecutive_auto_reply: Optional[int] = None,
-        human_input_mode: Optional[str] = "TERMINATE",
-        function_map: Optional[Dict[str, Callable]] = None,
-        code_execution_config: Union[Dict, Literal[False]] = False,
+        max_consecutive_auto_reply: int = MAX_CONSECUTIVE_AUTO_REPLY,
+        human_input_mode: HumanInputMode = "TERMINATE",
+        function_map: Dict[str, Callable] = {},
+        code_execution_config: Union[CodeExecutionConfig, Literal[False]] = False,
         llm_config: Optional[Union[Dict, Literal[False]]] = None,
         default_auto_reply: Union[str, Dict] = "",
         description: Optional[str] = None,
     ):
         """
         Args:
             name (str): name of the agent.
-            system_message (str or list): system message for the ChatCompletion inference.
+            system_message (str): system message for the ChatCompletion inference.
```

Comment:
I believe it is a mistake. According to the API docs for the system message on openai it is not valid to supply a list there, it can only be a string

Reply:
@BeibinLi can you confirm if list should be supported?

Reply:
No. It is not a mistake. For GPT-4V, it can take a list. For instance, if the system message contains an image, then it will be an array with 2-3 components, depends on the location of the image.

Reply:
Can you link me to the docs for gptv?

Reply:
We had a chat and I understand the situation better. I am going to revert the system message change in this PR, I think it is important and should be tackled separately.